3-extra/1-choose-your-own-adventure.js

```diff
@@ -53,6 +58,20 @@ let game = {
     //
     // Hint: the room objects have north/east/south/west methods which return
     // a new room object that is in the relevant direction.
+    let validDirections = ["north", "east", "south", "west"];
+    if (!validDirections.includes(direction)) {
+      console.log(
+        "Invalid direction entered. Please enter one of: north, south, east, west."
+      );
+      return;
+    }
+    let moveFunction = this.currentRoom[direction];
+    let newRoom = moveFunction();
+    if (!newRoom) {
+      console.log("Invalid direction");
+    } else {
+      this.currentRoom = newRoom;
+    }
```

Comment:
Well done Sobia :+1: 
Good understanding of array methods. Your code is clean and easy to read :)